hbase-server/src/main/java/org/apache/hadoop/hbase/wal/AbstractWALRoller.java

```diff
@@ -249,4 +247,43 @@ public void close() {
     running = false;
     interrupt();
   }
+
+  /**
+   * Independently control the roll of each wal. When use multiwal,
+   * can avoid all wal roll together. see HBASE-24665 for detail
+   */
+  protected class RollController {
+    private final WAL wal;
+    private final AtomicBoolean rollRequest;
+    private long lastRollTime;
+
+    RollController(WAL wal) {
+      this.wal = wal;
+      this.rollRequest = new AtomicBoolean(false);
+      this.lastRollTime = System.currentTimeMillis();
+    }
+
+    public void requestRoll() {
+      this.rollRequest.set(true);
+    }
+
+    public Map<byte[], List<byte[]>> rollWal(long now) throws IOException {
+      this.lastRollTime = now;
+      Map<byte[], List<byte[]>> regionsToFlush = wal.rollWriter(true);
+      this.rollRequest.set(false);
```

Comment:
Hey this is where we need to reset it even before we do actual roll work.. We were doing that.  We had a comment also.
// reset the flag in front to avoid missing roll request before we return from rollWriter.

Reply:
thank you very much! I also noticed this, but in order to avoid repeated roll, so fix to that. if use boolean, roll at most once, the impact can be ignored. Thanks for review

Reply:
Sorry did not get.  How AtomicBoolean is different compared to boolean wrt repeated roll possibility.  In the past we used to keep the Boolean state in a Map.  Now in this object.

Reply:
after reset flag, before roll is complete, if have a roll request, that will also cause an extra roll.

Reply:
Yes. That is happening now also.  We intentionally reset the boolean when we start roll itself.
// reset the flag in front to avoid missing roll request before we return from rollWriter.
So there is no change for that right now.

Reply:
yes, I go it